src/query/filter.go

```diff
@@ -0,0 +1,32 @@
+package query
+
+import (
+	"core"
+	"fmt"
+	"cli"
+)
+
+// Filter takes the list of BuildLabels and checks which ones match the label selectors passed in.
+// If no includeLabels are provided, the default behaviour is to include all labels.
+func Filter(graph *core.BuildGraph, labels core.BuildLabels, includeLabels []string, excludeLabels []string) {
+	matcher := func(target *core.BuildTarget) bool {
```

Comment:
I think `ExpandOriginalTargets` will do this for you if you set the top-level flags?